Detectors/TPC/workflow/src/TPCTimeSeriesSpec.cxx

```diff
@@ -1177,12 +1201,30 @@ class TPCTimeSeries : public Task
         const int nClITS = idxITSCheck ? tracksITS[idxITSTrack].getNClusters() : -1;
         const int chi2ITS = idxITSCheck ? tracksITS[idxITSTrack].getChi2() : -1;
         int typeSide = 2; // A- and C-Side cluster
-        if (track.hasASideClustersOnly()) {
+        if (trackFull.hasASideClustersOnly()) {
           typeSide = 0;
-        } else if (track.hasCSideClustersOnly()) {
+        } else if (trackFull.hasCSideClustersOnly()) {
           typeSide = 1;
         }
 
+        // check for TOF and propagate TPC track to TOF cluster
+        bool hasTOFCluster = (std::get<0>(idxTPCTrackToTOFCluster[iTrk]) != -1);
+        auto tofCl = hasTOFCluster ? tofClusters[std::get<0>(idxTPCTrackToTOFCluster[iTrk])] : o2::tof::Cluster();
+
+        float tpcYDeltaAtTOF = -999;
+        float tpcZDeltaAtTOF = -999;
+        if (hasTOFCluster) {
+          o2::track::TrackPar trackTmpOut(tracksTPC[iTrk].getParamOut());
+          if (trackTmpOut.rotate(o2::math_utils::sector2Angle(tofCl.getSector())) && propagator->propagateTo(trackTmpOut, tofCl.getX(), false, mMaxSnp, mFineStep, mMatType)) {
```

Comment:
Hello @noferini 
the propagation with the TPC track to TOF cluster is done here.
When we calculate the difference between TPC and TOF cluster in z (`tpcZDeltaAtTOF`) direction, we are also doing some correction, but I am not fully sure if this is correct:
`tpcZDeltaAtTOF = signSide * (o2::tpc::ParameterElectronics::Instance().ZbinWidth * trackFull.getTime0() - vertex.getTimeStamp().getTimeStamp()) * mVDrift - trackTmpOut.getZ() + tofCl.getZ();`

For the difference in y we are just doing `tpcYDeltaAtTOF = trackTmpOut.getY() - tofCl.getY();`

Reply:
@matthias-kleiner, I am not completely sure about the signs in tpcZDeltaAtTOF formula but for tpcYDeltaAtTOF there are no problems. I don't see any evident problem in the propagation call.
Do you have also tpcZDeltaAtTOF  vs phi/sector?

Reply:
Hi @noferini ,
yes sure it looks like this.
![tpcZDeltaAtTOF_544124_8kHz](https://github.com/AliceO2Group/AliceO2/assets/48915672/acacbee4-62c4-412c-89f6-bdeda05bb74d)

Reply:
It looks much better than Y residual.
What is the fine-step you are using in propagation? 2 cm (default)?
Does this play a role?